example-crates/wallet_rpc/src/main.rs

```diff
@@ -0,0 +1,91 @@
+use bdk::{
+    bitcoin::{Address, Network},
+    wallet::{AddressIndex, Wallet},
+    SignOptions,
+};
+use bdk_bitcoind_rpc::{
+    bitcoincore_rpc::{Auth, Client, RpcApi},
+    Emitter,
+};
+use bdk_file_store::Store;
+use std::str::FromStr;
+
+const DB_MAGIC: &str = "bdk-rpc-wallet-example";
+const SEND_AMOUNT: u64 = 5000;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args = std::env::args().collect::<Vec<_>>();
+    let db_path = std::env::temp_dir().join("bdk-rpc-example");
+    let db = Store::<bdk::wallet::ChangeSet>::new_from_path(DB_MAGIC.as_bytes(), db_path)?;
+
+    let external_descriptor = "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/0/*)";
+    let internal_descriptor = "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/1/*)";
+
+    let mut wallet = Wallet::new(
+        external_descriptor,
+        Some(internal_descriptor),
+        db,
+        Network::Testnet,
+    )?;
+
+    let address = wallet.get_address(AddressIndex::New);
+    println!("Generated Address: {}", address);
+
+    let balance = wallet.get_balance();
+    println!("Wallet balance before syncing: {} sats", balance.total());
+
+    let rpc_client = Client::new(&args[0], Auth::UserPass(args[1].clone(), args[2].clone()))?;
```

Comment:
If I recall, `args[0]` refers to the command executable, and subsequent arguments begin at index 1. I was only able to run this example after incrementing each of the subscripts.

bonus: You could check for the correct number of command line args early in the program and remind the user if something is missing.

```rust
if args.len() < 6 {
    println!("Usage: wallet_rpc <RPC_URL> <RPC_USER> <RPC_PASS> <LOOKAHEAD> <FALLBACK_HEIGHT>");
    std::process::exit(1);
}
```

Reply:
I like the bonus